src/boundaries/edgetracking.jl

```diff
@@ -0,0 +1,304 @@
+state space position of the updated edge point changes by less than `abstol` (in
+Euclidean distance) compared to the previous iteration. Convergence below `abstol` happens
+after sufficient iterations if the edge state is a saddle point. However, the edge state
+may also be an unstable limit cycle or a chaotic saddle. In these cases, the algorithm will
+never actually converge to a point but (after a burn-in) continue populating the set
+consituting the edge state by tracking along it.
+
+A central idea behind this algorithm is that basin boundaries are typically the stable
+manifolds of unstable sets, namely edge states or saddles. The flow along the basin boundary 
+will thus lead to these sets, and the iterative bisection neutralizes the unstable
+direction of the flow away from the basin boundary. If the system possesses multiple edge 
+states, the algorithm will find one of them depending on where the initial bisection locates
+the boundary.
+
+## Output
+
+Returns a data type `EdgeTrackingResults` containing five fields:
+* `edge::StateSpaceSet`: the pseudo-trajectory representing the tracked edge segment
+  (given by the average in state space between `track1` and `track2`)
+* `track1::StateSpaceSet`: the pseudo-trajectory tracking the edge within basin 1
+* `track2::StateSpaceSet`: the pseudo-trajectory tracking the edge within basin 2
+* `time::Vector`: time points of the above `StateSpaceSet`s
+* `bisect_idx::Vector`: indices of `time` at which a re-bisection occurred
+* `success::Bool`: indicates whether the edge tracking has been successful or not
+
+Sometimes, the AttractorMapper used in the algorithm may erroneously identify both states
+`u1` and `u2` with the same basin of attraction due to being very close to the basin
+boundary. If this happens, a warning is raised and `EdgeTrackingResults.success = false`.
+"""
+function edgetracking(ds::DynamicalSystem, attractors::Dict;
+    bisect_thresh=1e-6,
+    diverge_thresh=1e-5,
+    u1=collect(values(attractors))[1][1],
+    u2=collect(values(attractors))[2][1],
+    maxiter=100,
+    abstol=0.0,
+    T_transient=0.0,
+    tmax=Inf,
+    Δt=0.01,
+    ϵ_mapper=nothing,
+    show_progress=true,
+    kwargs...)
+    
+    pds = ParallelDynamicalSystem(ds, [u1, u2])
+    mapper = AttractorsViaProximity(ds, attractors, ϵ_mapper; kwargs...)
+    
+    edgetracking(pds, mapper;
+        bisect_thresh, diverge_thresh, maxiter, abstol, T_transient, Δt, tmax, show_progress)
+end;
+
+"""
+    edgetracking(pds::ParallelDynamicalSystem, mapper::AttractorMapper; kwargs...)
+Low-level function for running the edge tracking algorithm, see [`edgetracking`](@ref)
+for a description, keyword arguments and output type.
+
+`pds` is a `ParallelDynamicalSystem` with two states. The `mapper` must be an
+`AttractorMapper` of subtype `AttractorsViaProximity` or `AttractorsViaRecurrences`.
+"""
+function edgetracking(pds::ParallelDynamicalSystem, mapper::AttractorMapper;
```

Comment:
There should be a `verbose = true` keyword that enables the warnings, so that they can be silenced by the user with `verbose = false`. A user that anyways checks the success return code will not care about thrown wornings on the console. The keuword should also be used in `bisect_to_edge`.

Reply:
this keyword should also silence the success printing if false.

Reply:
Okay, done. Is there a smarter way to silence warnings than
```
if verbose
    @warn(...)
end
```
?